Makefile

```diff
@@ -0,0 +1,44 @@
+#
+# Top level Makefile to build Zenith and PostgreSQL
+#
+all: zenith postgres
+
+### Zenith Rust bits
+#
+# The 'postgres_ffi' depends on the Postgres headers.
+zenith: postgres-headers
+	cargo build
+
+### PostgreSQL parts
+tmp_install/build/config.status:
+	+@echo "Configuring postgres build"
+	mkdir -p tmp_install/build
+	(cd tmp_install/build && \
+	../../vendor/postgres/configure CFLAGS='-O0' --enable-debug --enable-cassert \
+	    --enable-depend --with-libxml --prefix=$(abspath tmp_install) > configure.log)
+
+# nicer alias for running 'configure'
+postgres-configure: tmp_install/build/config.status
+
+# Install the PostgreSQL header files into tmp_install/include
+postgres-headers: postgres-configure
+	$(MAKE) -C tmp_install/build/src/include MAKELEVEL=0 install
+
+# Compile and install PostgreSQL
+postgres: postgres-configure
+	$(MAKE) -j8 -C tmp_install/build MAKELEVEL=0 install
+
+postgres-clean:
+	$(MAKE) -C tmp_install/build/ MAKELEVEL=0 clean
+
+# This doesn't remove the effects of 'configure'.
+clean:
+	cd tmp_install/build && ${MAKE} clean
+	cargo clean
+
+# This removes everything
+distclean:
+	rm -rf tmp_install
```

Comment:
Shouldn't we add `cd tmp_install/build && ${MAKE} distclean` here?

Reply:
Not necessary when we do `rm -r tmp_install`. That removes the whole thing anyway.